Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -459,8 +460,29 @@ def _addToLog(self, logName, message):
     def evaluateCommand(self, cmd):
         rc = super().evaluateCommand(cmd)
         steps_to_add = self.follow_up_steps()
-        if rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash'):
-            steps_to_add += [ArchiveMinifiedBuiltProduct(), UploadMinifiedBuiltProduct(), TransferToS3(terminate_build=True)]
+
+        # Normal build archive
+        steps_to_add += [ArchiveBuiltProduct()]
+        if CURRENT_HOSTNAME in BUILD_WEBKIT_HOSTNAMES + TESTING_ENVIRONMENT_HOSTNAMES:
+            steps_to_add.extend([
+                GenerateS3URL(f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}"),
+                UploadFileToS3(f"WebKitBuild/{self.getProperty('configuration')}.zip", links={self.name: 'Archive'}),
+            ])
+        else:
+            # S3 might not be configured on local instances, achieve similar functionality without S3.
+            steps_to_add.extend([UploadBuiltProduct(), TransferToS3(terminate_build=True)])
+
+        # Minified build archive
+        if self.getProperty('platform') in self.APPLE_PLATFORMS or rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash'):
```

Comment:
should it be `or` or `and`

Reply:
Based on what I read in the original `BuildFactory`, we want a minified build archive when we're on an Apple platform. And then combined with the condition in `evaluateCommand`, we get:

`(if platform in apple) OR (rc in success, warnings AND user_provided_git_hash)` which I believe is correct.